src/presenter/content-presenter.js

```diff
@@ -19,69 +19,126 @@ import { render } from 'utils';
+
+  #openPopup = (filmId) => {
+    this.#renderPopup(filmId);
+    this.#bodyElement.classList.add(bodyHideOverflowClass);
+    document.addEventListener('keydown', this.#onEscKeyDown);
   };
 
-  renderPopup = (popupFilmId) => {
-    this.popupFilm = this.filmModel.getFilm(popupFilmId);
-    this.popupComponent = new PopupView();
-    this.popupCommentListComponent = new PopupCommentListView();
-    this.popupNewCommentComponent = new PopupNewCommentView();
-    this.popupTopContainerComponent = new PopupTopContainerView(this.popupFilm);
-    render(this.popupTopContainerComponent, this.popupComponent.getElement());
+  #closePopup = () => {
+    this.#popupComponent.element.remove();
+    this.#popupComponent.removeElement();
+    this.#bodyElement.classList.remove(bodyHideOverflowClass);
+  };
+
+  #onEscKeyDown = (evt) => {
+    if (evt.key === 'Escape') {
+      evt.preventDefault();
+      this.#closePopup();
+      document.removeEventListener('keydown', this.#onEscKeyDown);
+    }
+  };
 
-    this.popupBottomContainerComponent = new PopupBottomContainerView(
-      this.popupFilm.commentsIds.length
+  #renderPopup = (popupFilmId) => {
+    this.#popupFilm = this.#filmModel.getFilm(popupFilmId);
+    this.#popupComponent = new PopupView();
+    this.#popupCommentListComponent = new PopupCommentListView();
+    this.#popupNewCommentComponent = new PopupNewCommentView();
+    this.#popupTopContainerComponent = new PopupTopContainerView(this.#popupFilm);
+    render(this.#popupTopContainerComponent, this.#popupComponent.element);
+
+    this.#popupBottomContainerComponent = new PopupBottomContainerView(
+      this.#popupFilm.commentsIds.length
     );
 
-    this.filmComments = this.commentModel.getFilmComments(popupFilmId, this.filmModel);
+    this.#filmComments = this.#commentModel.getFilmComments(popupFilmId, this.#filmModel);
 
-    this.filmComments.forEach((comment) => {
-      render(new PopupCommentItemView(comment), this.popupCommentListComponent.getElement());
+    this.#filmComments.forEach((comment) => {
+      render(new PopupCommentItemView(comment), this.#popupCommentListComponent.element);
     });
 
-    render(this.popupCommentListComponent, this.popupBottomContainerComponent.getElement());
-    render(this.popupNewCommentComponent, this.popupBottomContainerComponent.getElement());
-    render(this.popupBottomContainerComponent, this.popupComponent.getElement());
-    render(this.popupComponent, this.siteFooterElement, 'afterend');
+    render(this.#popupCommentListComponent, this.#popupBottomContainerComponent.element);
+    render(this.#popupNewCommentComponent, this.#popupBottomContainerComponent.element);
+    render(this.#popupBottomContainerComponent, this.#popupComponent.element);
+    render(this.#popupComponent, this.#siteFooterElement, 'afterend');
+
+    this.#popupTopContainerComponent.element.querySelector(popupCloseSelector).addEventListener(
+      'click', () => {
+        this.#closePopup();
+      }
```

Comment:
```suggestion
      'click', this.#closePopup
      }
```

зачем создавать новую анонимную функцию, если можно передать ссылку на метод закрытия попапа.